accepted/00000-ptf-repos-in-uyuni-2.md

```diff
@@ -0,0 +1,179 @@
+- Feature Name: Use PTFs in Uyuni
+- Start Date: 2021-06-21
+
+# Summary
+[summary]: #summary
+
+SUSE want to provide customers PTFs (Product Temporary Fixes) in repositories.
+This RFC is about how PTFs will look like and how we can manage and install them on clients.
+
+# Motivation
+[motivation]: #motivation
+
+PTFs are Product Temporary Fixes for Service Requests opened by SUSE customers.
+The resulting packages are the last released maintenance update plus a patch which fixes the actual issue.
+
+PTFs are made for dedicated customers. This means every customer has different PTF repositories, with different content and fixes for different service requests.
+
+When we provide these PTFs in SUSE Manager as channels, it is easier to roll out the fixes to multiple clients.
+We need to take care that the PTFs can be installed, updated and been replaced by a regular maintenance update which contains the fix.
+
+Sometimes SUSE provide also TEST packages. The difference is, that they are just meant for testing on a single system to better identify the problem and gather logfiles.
+These packages are not supported and should not rolled out on multiple systems.
+It is planned to have them in separate channels. We need to provide a best practice to work with them.
+We need to be able to install and update them. Replacing them by a PTF or a maintenance update needs to be possible.
+
+# Detailed design
+[design]: #detailed-design
+
+## General PTF design for SUSE OSes
+
+A ptf release consists of multiple rpm packages plus one
+master "ptf" package. The master ptf package has two
+purposes:
+
+- it allows to easily query which ptfs are installed
+  in the system
+- it makes sure that all of the installed rpm packages
+  come from the same release, i.e. there is no mix up
+  between multiple releases are non-ptf package
+
+To do this the master ptf package contains the following
+elements (incident 1234, first release):
+
+    Name: ptf-1234
+    Version: 1
+    Release: 1                 # always 1
+    Provides: ptf() = 1234-1
+    Requires: (pkg1 = pkg1EVR if pkg1)
+    Requires: (pkg2 = pkg1EVR if pkg2)
+    ...
+
+I.e. if pkg1 is installed it must be installed with version
+pkg1EVR. We do not use `Conflict: pkg1 != pkg1EVR` for SUSE because that
+would not work with multiversion packages like the kernel.
+But for Non-SUSE systems and packages, which do not support multiversion
+packages, we could generate `Conflicts` dependencies.
```

Comment:
In this sentence I guess that "we" refers to SUSE PTF authors rather than Uyuni or SUMa, right?